src/charm.py

```diff
@@ -309,26 +313,20 @@ def bootstrap_cluster(self, event: ops.framework.EventBase) -> None:
             if error_already_exists not in e.stderr:
                 raise e
 
-    def configure_ceph(self) -> None:
-        """Configure Ceph.
+    def configure_ceph(self, event) -> None:
+        """Configure Ceph."""
+        if not self.ready_for_service():
+            event.defer()
+            return
 
-        Set mon_allow_pool_size_one to true
-        """
-        cmds = [
-            ["ceph", "config", "set", "global", "mon_allow_pool_size_one", "true"],
-            ["ceph", "config", "set", "global", "osd_pool_default_size", "1"],
-        ]
+        default_rf = self.model.config.get("default-pool-size")
         try:
-            for cmd in cmds:
-                logger.debug(f'Running command {" ".join(cmd)}')
-                process = subprocess.run(
-                    cmd, capture_output=True, text=True, check=True, timeout=60
-                )
-                logger.debug(
-                    f"Command finished. stdout={process.stdout}, " f"stderr={process.stderr}"
-                )
-        except (subprocess.CalledProcessError, subprocess.TimeoutExpired) as e:
-            logger.warning(e.stderr)
+            microceph.set_pool_size("", str(default_rf))
+        except subprocess.CalledProcessError as e:
+            if "unknown command" in e.stderr:
+                if default_rf != 3:
+                    event.fail("cannot set pool size: command not supported by microceph")
+                return
```

Comment:
Some comments on the error handling would be awesome.

Reply:
ack.